src/structures/game/Matrix.js

```diff
@@ -0,0 +1,26 @@
+module.exports = class Matrix extends Array {
+  constructor (rows, columns) {
+    super()
+
+    this.rows = rows
+    this.columns = columns
+  }
+
+  populate (callback = () => 0) {
+    for (let i = 0; i < this.rows; i++) {
+      this[i] = []
+
+      for (let j = 0; j < this.columns; j++) {
+        this[i][j] = callback()
```

Comment:
is callback the right name? also would be cool to put the location at the function, so we could use like
```js
matrix.populate((i, j) => i * j)
```

Reply:
I forgot to put location on function and about the name, I really don't know other name I can put.

Reply:
I will keep `callback` name, because `Array.prototype.map` receive `callback` too